contentgrid-spring-data-pagination/src/main/java/com/contentgrid/spring/data/pagination/web/ContentGridSpringDataPaginationWebConfiguration.java

```diff
@@ -11,15 +13,20 @@
 @Configuration(proxyBeanMethods = false)
 public class ContentGridSpringDataPaginationWebConfiguration {
 
+
     @Bean
     static BeanPostProcessor replacePagedResourceAssemblerBeanPostProcessor(
-            @Lazy HateoasPageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver
+            @Lazy HateoasPageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver,
+            ObjectProvider<CursorEncoder> cursorEncoder
     ) {
         return new BeanPostProcessor() {
             @Override
             public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
                 if (bean instanceof PagedResourcesAssembler<?>) {
-                    return new ItemCountPageResourceAssembler<>(pageableHandlerMethodArgumentResolver);
+                    return new ItemCountPageResourceAssembler<>(
+                            pageableHandlerMethodArgumentResolver,
+                            org.springframework.data.util.Lazy.of(cursorEncoder::getIfUnique)
```

Comment:
Won't this possibly result in NullPointers when there is no CursorEncoder or when there are multiple?

Reply:
There is a check at the usage site, falling back to an empty `CursorPageMetadata` if there is no `CursorEncoder` in the `Lazy`, so no null pointer exceptions.

https://github.com/xenit-eu/contentgrid-spring/blob/927e93c6c7f468bf2188ebee39b61a3df438be6b/contentgrid-spring-data-pagination/src/main/java/com/contentgrid/spring/data/pagination/web/ItemCountPageResourceAssembler.java#L74-L79

The result of a `CursorPageMetadata` with null values is that no `prev_cursor` or `next_cursor` fields get rendered.

You did find an issue here though: if there are multiple `CursorEncoder`s (which *should not happen*), there won't be any cursors rendered either. I think we prefer an actual exception in that case indicating that there is an error.

Unfortunately, I don't think there is a way to make the error appear during startup, we don't want to have any bean eagerly-initialized for injection in a `BeanPostProcessor`.

So I think we should replace the `getIfUnique` with `getIfAvailable` (which will throw when there are duplicate beans without any primary or priority difference)

```suggestion
                            org.springframework.data.util.Lazy.of(cursorEncoder::getIfAvailable)
```